draco/asp/generate.lp

```diff
@@ -38,10 +36,14 @@ not_required((encoding,aggregate)).
 not_required((encoding,field)).
 not_required((encoding,stack)).
 
-% generator(scale) generate scales such that their ids and channels corresponde to the encodings in each view.
-{ entity(scale,V,E) } = 1 :- entity(mark,V,M), entity(encoding,M,E), attribute((encoding,channel),E,C).
-attribute((scale,channel),S,C) :- entity(scale,V,S), attribute((encoding,channel),S,C).
+% @generator(scale) generate scales for all encoding channels from each view.
+% count the total number of channels of the marks in the same view.
+count_channels(V,N) :- N = #count{ C : domain(encoding_channel,V,C) }, entity(view,_,V).
```

Comment:
We should try to avoid putting the count into a fact. The problem with doing that is that the grounding can get large. Could we reuse the encoding id somehow in the scale id generation? Id not, that's okay.

Reply:
I tried using the encoding ids before which resulted the issue #332: some scale entities are included in the partial spec but ignored by the generator; it instead generates all new scales based the encodings. Do we always assume scale entities are not in the spec? If so, we are safe to use the encoding ids.

Reply:
What do you mean by "Do we always assume scale entities are not in the spec"?

Reply:
I think they can but I would not expect them to contain scales in most cases.

Reply:
@domoritz 
I just saw that the grounding was much better using the count than using encoding ids.
histogram: 5184 vs 7496
scatter: 8938 vs 12120

Should I revert it back?